src/screens/DiscoveryScreenV2/fragment/TopicFragment.js

```diff
@@ -93,33 +110,77 @@ const TopicFragment = ({
-        key={`followedTopic-${index}`}
-        onPressBody={() => __handleOnTopicPress(item)}
-        isHashtag
-        item={{
-          name: item.name,
-          image: item.profile_pic_path,
-          isunfollowed: item.user_id_follower === null,
-          description: null
-        }}
-      />
-    </View>
-  );
+  const __renderDiscoveryItem = (from, key, item, index) => {
+    return (
+      <View key={`${key}-${index}`} style={styles.domainContainer}>
+        {route.name === 'Followings' && item.user_id_follower !== null && (
+          <DomainList
+            handleSetFollow={() => handleFollow(from, true, item, index)}
+            handleSetUnFollow={() => handleFollow(from, false, item, index)}
+            key={`followedTopic-${index}`}
+            isCommunity={true}
+            onPressBody={() => __handleOnTopicPress(item)}
+            item={{
+              name: item.name,
+              image: item.icon_path,
+              isunfollowed: !item.following,
+              description: null
+            }}
+            DefaultImage={TopicsProfilePictureEmptyState}
+          />
+        )}
+        {route.name !== 'Followings' && (
+          <DomainList
+            handleSetFollow={() => handleFollow(from, true, item, index)}
+            handleSetUnFollow={() => handleFollow(from, false, item, index)}
+            key={`followedTopic-${index}`}
+            isCommunity={true}
+            onPressBody={() => __handleOnTopicPress(item)}
+            item={{
+              name: item.name,
+              image: item.icon_path,
+              isunfollowed: !item.following,
+              description: null
+            }}
+            DefaultImage={TopicsProfilePictureEmptyState}
+          />
+        )}
+      </View>
+    );
+  };
 
   const __renderTopicItems = () => {
+    const followingTopics = [];
+    const unfollowingTopics = [];
+
+    topics.forEach((item) => {
+      if (item.user_id_follower) {
+        followingTopics.push(item);
+      } else {
+        unfollowingTopics.push(item);
+      }
+    });
     if (isFirstTimeOpen) {
       return [
-        <DiscoveryTitleSeparator key="topic-title-separator" text="Suggested Communities" />
-      ].concat(
-        topics.map((item, index) =>
+        followingTopics.map((item, index) =>
           __renderDiscoveryItem(FROM_FOLLOWED_TOPIC_INITIAL, 'followedTopicDiscovery', item, index)
         )
-      );
+      ]
+        .concat([
+          route.name !== 'Followings' && (
+            <DiscoveryTitleSeparator key="topic-title-separator" text="Suggested Communities" />
+          )
+        ])
+        .concat(
+          unfollowingTopics.map((item, index) =>
+            __renderDiscoveryItem(
+              FROM_FOLLOWED_TOPIC_INITIAL,
+              'followedTopicDiscovery',
+              item,
+              index + followingTopics.length
+            )
+          )
+        );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `__renderTopicItems` function has been updated to handle different rendering based on the route name and the follower status of each topic. This change increases the complexity of the function, but it seems necessary for the new feature.
```diff
- const __renderTopicItems = () => {
-  if (isFirstTimeOpen) {
-    return [
-     <DiscoveryTitleSeparator key="topic-title-separator" text="Suggested Communities" />
-   ].concat(
-     topics.map((item, index) =>
-        __renderDiscoveryItem(FROM_FOLLOWED_TOPIC_INITIAL, 'followedTopicDiscovery', item, index)
-      )
-   );
-  }
+ const __renderTopicItems = () => {
+  const followingTopics = [];
+  const unfollowingTopics = [];
+
+  topics.forEach((item) => {
+    if (item.user_id_follower) {
+      followingTopics.push(item);
+    } else {
+      unfollowingTopics.push(item);
+    }
+  });
+  if (isFirstTimeOpen) {
+     return [
+      followingTopics.map((item, index) =>
+         __renderDiscoveryItem(FROM_FOLLOWED_TOPIC_INITIAL, 'followedTopicDiscovery', item, index)
+       )
+    ]
+      .concat([
+        route.name !== 'Followings' && (
+          <DiscoveryTitleSeparator key="topic-title-separator" text="Suggested Communities" />
+        )
+      ])
+      .concat(
+        unfollowingTopics.map((item, index) =>
+          __renderDiscoveryItem(
+            FROM_FOLLOWED_TOPIC_INITIAL,
+            'followedTopicDiscovery',
+            item,
+            index + followingTopics.length
+          )
+        )
+      );
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->